src/converter/one/proto/Proto.java

```diff
@@ -82,7 +82,7 @@ public void writeLong(long n) {
         int length = n == 0 ? 1 : (70 - Long.numberOfLeadingZeros(n)) / 7;
         ensureCapacity(length);
 
-        while (n > 0x7f) {
+        while (n & 0x7f != n) {
```

Comment:
A canonical way to check if an integer has higher bits set is
```
(n & ~0x7f) != 0
```
or
```
(n >>> 7) != 0
```

Reply:
I changed the check based on your suggestion.